src/Functions/FunctionsDecimalArithmetics.h

```diff
@@ -0,0 +1,445 @@
+#pragma once
+#include <type_traits>
+#include <Core/AccurateComparison.h>
+
+#include <DataTypes/DataTypesDecimal.h>
+#include <Columns/ColumnsNumber.h>
+#include <Functions/IFunction.h>
+#include <Functions/FunctionHelpers.h>
+#include <Functions/castTypeToEither.h>
+#include <IO/WriteHelpers.h>
+
+#include <Common/logger_useful.h>
+#include <Poco/Logger.h>
+#include <Loggers/Loggers.h>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int DECIMAL_OVERFLOW;
+    extern const int ILLEGAL_COLUMN;
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+    extern const int ILLEGAL_DIVISION;
+}
+
+
+struct DecimalOpHelpers
+{
+    static std::vector<UInt8> multiply(const std::vector<UInt8> & num1, const std::vector<UInt8> & num2)
```

Comment:
Let's add some comments about how these functions work and what is the expected number representation.

Also, it's unclear why we use Int32 and UInt16 for sizes and how it's guaranteed to fit (probably add some safety checks?).